### DISCUSSION TITLE
Hackathon setup

### DISCUSSION THREAD

**Discussion by pz-max** [pypsa-meets-africa/pypsa-africa-hackathon#2] [General]
I drafted regarding the upcoming hackathon (01.12., 12:00 UK time) a layout of the event in the ReadMe. @davide-f , @mnm-matin , and @LukasFrankenQ what are your thoughts?

**Comment by pz-max** [comment]
I would love seeing @mnm-matin sharing some insights/experiences as the most ambitious outlier in the team :1st_place_medal: :)

**Comment by davide-f** [comment]
Congratulations and thanks for drafting it up!
I propose a possible outline that complements yours:
1) The PyPSA Earth mission: the mission and the features we want from the code [basically three slides from past presentations]
2) PyPSA Africa
2.1) mission
2.2) Architecture
2.3) Requirements (Python, "VSCode", conda, solver): briefly introduce them and check if all have them installed
2.3.1) Github
2.3.2) VScode
2.3.3) conda
2.3.4) Solver [briefly]
2.4) the workflow
3) Practice [the section already described]

Regarding the hackaton, the new section cannot be a fork of pypsa-africa in the current stage: we cannot on-demand develop the retrieve-databundle to everyone, unless we succeed in reducing the size of the files.
However, we could say the audience to copy on real-time the folders data/resources/cutouts from the other repository to this one; snakemake will raise also errors that can be tackled and be a first exercise for them.

**Comment by pz-max** [comment]
Hi @davide-f,
Very good, I implemented all points in the readme.
I think we should avoid the previous download. Maybe we can store the data for a 2 country case [egypt, sudan] in the repo directly. This would require creating a cutout, OSM data, for this country. We would need such a minimal example for the CI anyways.

**Reply by davide-f** [reply]
I agree that we need the sample case for CI, however, for CI the sample should be really minimal, like Togo+Benin, or Sierra Leone + Liberia or alike: the smaller and compact the better.
We may prepare a notebook to draw desired cutouts so that we can be flexible.
For this example, maybe Nigeria could be nice if it is not too big.

**Comment by LukasFrankenQ** [comment]
Cheers!
Yeah, I think the draft looks great! I particularly like the 7-step thing with git, that xqdc comic captures my previous approach perfectly ;) But as far as I can see the only way people are getting their hands dirty is in terms of git, I like the idea of having a mini-simulation that allows people to also play around with a power model!

**Reply by davide-f** [reply]
@LukasFrankenQ, In the "Executiion, Snakemake and debugging" section, I believe @pz-max intended to work on pypsa-africa and execute/debug the snakemake workflow.
Maybe few words for clarifying may help

**Reply by pz-max** [reply]
### Installation [ACTION] (20-30min) -- Max/Davide/Lukas
- Together (clone already done above, continiue with environment etc. (homework?))
**- brief intro to the analyis [Jupyer script, import network -> analyse, series vs integer/float]** Just added this

**Reply by pz-max** [reply]
But yeah the whole draft needs to be rewritten (just shared some thoughts). Think we discuss it later and then go ahead :)

**Comment by davide-f** [comment]
@pz-max Max, I think we should clarify in the installation process to rename the pypsa-africa-hackaton package as "pypsa-africa" for full compatibility.
As an alternative, we may change the _sets_path_to_root script to find a folder that "starts with" the specified string, yet, that's a trick

In the scripts, there is the _sets_path_to_root("pypsa-africa") that may break if we rename the folder differently.

**Comment by pz-max** [comment]
Makes sense! Think option one is much easier. We also need to fix environments so that we don’t need to update the hackathon stuff all the time.

Having on future package labels will make life much easier

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by pz-max** [comment]
Hi @davide-f , @LukasFrankenQ  and @mnm-matin . 
FYI, I cleaned up some stuff. All should work now. 
(also your example @LukasFrankenQ so nothing to worry atm :) )
